### ISSUE TITLE
UDDataGrid sorting is not working

### ISSUE THREAD

**Issue by timoline** [ironmansoftware/powershell-universal#3354]
### Version

4.2.21

### Severity

Low

### Environment

msi

### Steps to Reproduce

I am experincing some problems with sorting in 
**New-UDDataGrid**
DefaultSortColumn and DefaultSortDirection does not sort (or does not sort correct)

**New-UDDataGridColumn**
SortingOrder  does not sort (or does not sort correct)

### Expected behavior

```console
Sort Columns
```

### Actual behavior

```console
does not sort (or does not sort correct)
```

**Comment by timoline** [comment]
And here is another sorting bug (see comments)

UDDataGrid sort error #3210

**Comment by adamdriscoll** [comment]
I can't reproduce some of this so I will need some examples. Here is the current code I'm working with. 

```powershell
New-UDApp -Title "component test dash" -Content {
  New-UDDataGrid -LoadRows {
      $Data = @(
          @{ Name = 'A'; Number = "Z11"}
          @{ Name = 'B'; Number = "Z10"}
          @{ Name = 'C'; Number = "Z9"}
          @{ Name = 'D'; Number = "Z8"}
          @{ Name = 'E'; Number = "Z7"}
          @{ Name = 'F'; Number = "Z6"}
          @{ Name = 'H'; Number = "Z5"}
          @{ Name = 'I'; Number = "Z4"}
          @{ Name = 'J'; Number = "Z3"}
          @{ Name = 'K'; Number = "Z2"}
          @{ Name = 'L'; Number = "Z1"}
      )
      $Data | Out-UDDataGridData -Context $EventData -TotalRows $Rows.Length
  } -Columns @(
      New-UDDataGridColumn -Field name -Sortable 
      New-UDDataGridColumn -Field number -Sortable 
  ) -AutoHeight $true #-DefaultSortDirection asc -DefaultSortColumn number
}
```

If I don't do anything, the grid is unsorted. 

![image](https://github.com/ironmansoftware/issues/assets/1256531/8ab5c384-af36-4d78-8216-eafac186f39f)

If I sort the name column ascending, it is sorted properly. 

![image](https://github.com/ironmansoftware/issues/assets/1256531/004cef87-e474-4cef-8094-4e140e15255e)

If I sort descending, it sorts properly. 

![image](https://github.com/ironmansoftware/issues/assets/1256531/3027db64-4aa7-46d1-81d3-22455b4e3146)

If I click one more time, it goes to unsorted properly. 

![image](https://github.com/ironmansoftware/issues/assets/1256531/1e60b0fa-ca05-4de0-b71b-be21828e563d)

For the default sort column name and direction, that works as long as I make the property name lower case. 

```powershell
 ) -AutoHeight $true -DefaultSortDirection asc -DefaultSortColumn number
```

![image](https://github.com/ironmansoftware/issues/assets/1256531/ccdcbd41-b94e-4d7a-a1b1-863a9b01ece5)

Sorting order doesn't seem to have an effect.

**Comment by timoline** [comment]
thx for the feedback, I tested it again,
-  header sorting is indeed working, like you said
- DefaultSortColumn and DefaultSortDirection is also working when indeed property is in lowercase (didn't know that, any other then it's broken)
- SortingOrder isn't worknig

**Comment by adamdriscoll** [comment]
Cool. We can fix the issue with the DefaultSoftColumn case sensitivity and look into the sorting order.

**Comment by adamdriscoll** [comment]
I've added some docs about -SortingOrder because I didn't even remember what it was intended for: https://docs.powershelluniversal.com/userinterfaces/components/data-display/data-grid#columns 

It does seem to be working if you have the casing right. For example, if you do this, it won't have an unsorted state. 

```
-SortingOrder @("asc", "desc") 
```

It suffers from the same case sensitivity as -DefaultSortColumn. These are fixed and will be in v4.3.
